src/lsm/fuzz.zig

```diff
@@ -0,0 +1,486 @@
+        env.forest = undefined;
+        env.forest_exists = false;
+
+        env.state = .init;
+    }
+
+    fn deinit(env: *Environment) void {
+        assert(env.state != .uninit);
+        defer env.state = .uninit;
+
+        if (env.forest_exists) {
+            env.forest.deinit(allocator);
+            env.forest_exists = false;
+        }
+        env.grid.deinit(allocator);
+        env.superblock.deinit(allocator);
+        env.message_pool.deinit(allocator);
+    }
+
+    fn tick(env: *Environment) !void {
+        env.grid.tick();
+        env.storage.tick();
+    }
+
+    pub fn format(storage: *Storage) !void {
+        var env: Environment = undefined;
+
+        try env.init(storage);
+        defer env.deinit();
+
+        assert(env.state == .init);
+        env.superblock.format(superblock_format_callback, &env.superblock_context, .{
+            .cluster = cluster,
+            .replica = replica,
+            .size_max = size_max,
+        });
+
+        while (true) {
+            switch (env.state) {
+                .init => try env.tick(),
+                .formatted => break,
+                else => unreachable,
+            }
+        }
+    }
+
+    fn superblock_format_callback(superblock_context: *SuperBlock.Context) void {
+        const env = @fieldParentPtr(@This(), "superblock_context", superblock_context);
+        assert(env.state == .init);
+        env.state = .formatted;
+    }
+
+    pub fn open(env: *Environment) !void {
+        assert(env.state == .init);
+        env.superblock.open(superblock_open_callback, &env.superblock_context);
+
+        while (true) {
+            switch (env.state) {
+                .init, .superblock_open => try env.tick(),
+                .forest_open => break,
+                else => unreachable,
+            }
+        }
+    }
+
+    fn superblock_open_callback(superblock_context: *SuperBlock.Context) void {
+        const env = @fieldParentPtr(@This(), "superblock_context", superblock_context);
+        assert(env.state == .init);
+        env.state = .superblock_open;
+
+        env.forest = Forest.init(allocator, &env.grid, node_count, forest_options) catch unreachable;
+        env.forest_exists = true;
+        env.forest.open(forest_open_callback);
+    }
+
+    fn forest_open_callback(forest: *Forest) void {
+        const env = @fieldParentPtr(@This(), "forest", forest);
+        assert(env.state == .superblock_open);
+        env.state = .forest_open;
+    }
+
+    pub fn checkpoint(env: *Environment) !void {
+        assert(env.state == .forest_open);
+        env.state = .forest_checkpointing;
+        env.forest.checkpoint(forest_checkpoint_callback);
+
+        while (true) {
+            switch (env.state) {
+                .forest_checkpointing, .superblock_checkpointing => try env.tick(),
+                .forest_open => break,
+                else => unreachable,
+            }
+        }
```

Comment:
Maybe cleaner as
```suggestion
        while (env.state == .forest_checkpointing or env.state == .superblock_checkpointing) {
            try env.tick();
        }
        assert(env.state == .forest_open);
```

or even
```suggestion
        while (env.state == .forest_checkpointing) try env.tick();
        while (env.state == .superblock_checkpointing) try env.tick();
        assert(env.state == .forest_open);
```